File: venered-kmp/shared/src/commonMain/kotlin/com/venered/social/presentation/theme/VeneredTheme.kt
package com.venered.social.presentation.theme

import androidx.compose.ui.graphics.Color

/**
 * Paleta de colores centralizada para toda la app
 * Se comparte entre Android e iOS
 */
object VeneredColors {
    // ============================================================
    // DARK THEME
    // ============================================================
    
    // Primario
    val DarkPrimary = Color(0xFF6366F1)              // Indigo vibrante
    val DarkPrimaryVariant = Color(0xFF4F46E5)      // Indigo oscuro
    
    // Secundario
    val DarkSecondary = Color(0xFFEC4899)            // Rosa vibrante
    val DarkSecondaryVariant = Color(0xFFDB2777)    // Rosa oscuro
    
    // Fondos
    val DarkBackground = Color(0xFF0A0A0A)           // Casi negro
    val DarkSurface = Color(0xFF171717)              // Gris muy oscuro
    val DarkSurfaceVariant = Color(0xFF1E1028)      // Gris con tinte morado
    
    // Textos
    val DarkTextPrimary = Color(0xFFFFFFFF)          // Blanco
    val DarkTextSecondary = Color(0xFFCACACB)        // Gris claro
    val DarkTextTertiary = Color(0xFF808080)        // Gris medio
    
    // Estados
    val DarkError = Color(0xFFFF6B6B)                // Rojo
    val DarkSuccess = Color(0xFF51CF66)              // Verde
    val DarkWarning = Color(0xFFFFA94D)              // Naranja
    
    // ============================================================
    // LIGHT THEME
    // ============================================================
    
    // Primario
    val LightPrimary = Color(0xFF4F46E5)             // Indigo
    val LightPrimaryVariant = Color(0xFF3730A3)     // Indigo oscuro
    
    // Secundario
    val LightSecondary = Color(0xFFEA580C)           // Naranja
    val LightSecondaryVariant = Color(0xFFD64A0A)   // Naranja oscuro
    
    // Fondos
    val LightBackground = Color(0xFFF5F8FF)          // Azul muy claro
    val LightSurface = Color(0xFFFFFFFF)             // Blanco
    val LightSurfaceVariant = Color(0xFFEEF2FF)     // Azul claro
    
    // Textos
    val LightTextPrimary = Color(0xFF0F172A)         // Casi negro
    val LightTextSecondary = Color(0xFF475569)       // Gris oscuro
    val LightTextTertiary = Color(0xFF94A3B8)       // Gris
    
    // Estados
    val LightError = Color(0xFFEF4444)               // Rojo
    val LightSuccess = Color(0xFF22C55E)             // Verde
    val LightWarning = Color(0xFFF59E0B)             // Naranja
    
    // ============================================================
    // NEUTRAL (Compartidos)
    // ============================================================
    
    val Transparent = Color(0x00000000)
    val Black = Color(0xFF000000)
    val White = Color(0xFFFFFFFF)
    
    // Bordes y divisores
    val DarkBorder = Color(0xFF2D2D2D)
    val LightBorder = Color(0xFFDCE5F6)
}

/**
 * Espaciado centralizado (dp values)
 */
object VeneredSpacing {
    const val ExtraSmall = 4          // 4.dp
    const val Small = 8               // 8.dp
    const val Medium = 12             // 12.dp
    const val Large = 16              // 16.dp
    const val ExtraLarge = 24         // 24.dp
    const val Huge = 32               // 32.dp
    const val Massive = 48            // 48.dp
}

/**
 * Tamaños de fuentes centralizados (sp)
 */
object VeneredTypography {
    // Display
    const val DisplayLargeSize = 48
    const val DisplayMediumSize = 40
    const val DisplaySmallSize = 32
    
    // Headline
    const val HeadlineLargeSize = 28
    const val HeadlineMediumSize = 24
    const val HeadlineSmallSize = 20
    
    // Title
    const val TitleLargeSize = 20
    const val TitleMediumSize = 16
    const val TitleSmallSize = 14
    
    // Body
    const val BodyLargeSize = 16
    const val BodyMediumSize = 14
    const val BodySmallSize = 12
    
    // Label
    const val LabelLargeSize = 14
    const val LabelMediumSize = 12
    const val LabelSmallSize = 11
}

/**
 * Radio de esquinas (border radius)
 */
object VeneredCornerRadius {
    const val ExtraSmall = 4
    const val Small = 8
    const val Medium = 12
    const val Large = 16
    const val ExtraLarge = 20
    const val Circle = 50  // Para elementos circulares
}

/**
 * Elevación (shadow depth)
 */
object VeneredElevation {
    const val None = 0
    const val Low = 2
    const val Medium = 4
    const val High = 8
    const val VeryHigh = 12
}

/**
 * Opacidades
 */
object VeneredOpacity {
    const val Disabled = 0.38f
    const val Hovered = 0.08f
    const val Focused = 0.12f
    const val Pressed = 0.12f
    const val DraggedAlpha = 0.16f
}

/**
 * Duraciones de animación (ms)
 */
object VeneredDuration {
    const val VeryShort = 100
    const val Short = 200
    const val Medium = 300
    const val Long = 500
    const val VeryLong = 1000
}

File: venered-kmp/iosApp/VeneredTheme.swift
import SwiftUI

// COLORS - Sincronizados con Kotlin
struct VeneredColors {
    // DARK THEME
    static let darkPrimary = Color(red: 0.38, green: 0.40, blue: 0.95)        // #6366F1
    static let darkSecondary = Color(red: 0.92, green: 0.28, blue: 0.60)      // #EC4899
    static let darkBackground = Color(red: 0.04, green: 0.04, blue: 0.04)     // #0A0A0A
    static let darkSurface = Color(red: 0.09, green: 0.09, blue: 0.09)        // #171717
    static let darkTextPrimary = Color(red: 1.0, green: 1.0, blue: 1.0)       // #FFFFFF
    static let darkTextSecondary = Color(red: 0.79, green: 0.79, blue: 0.80)  // #CACACA
    static let darkError = Color(red: 1.0, green: 0.42, blue: 0.42)           // #FF6B6B
    
    // LIGHT THEME
    static let lightPrimary = Color(red: 0.31, green: 0.27, blue: 0.90)       // #4F46E5
    static let lightSecondary = Color(red: 0.92, green: 0.34, blue: 0.05)     // #EA580C
    static let lightBackground = Color(red: 0.96, green: 0.97, blue: 1.0)     // #F5F8FF
    static let lightSurface = Color(red: 1.0, green: 1.0, blue: 1.0)          // #FFFFFF
    static let lightTextPrimary = Color(red: 0.06, green: 0.09, blue: 0.16)   // #0F172A
    static let lightTextSecondary = Color(red: 0.28, green: 0.33, blue: 0.41) // #475569
    static let lightError = Color(red: 0.94, green: 0.27, blue: 0.27)         // #EF4444
}

// SPACING - Sincronizado con Kotlin
struct VeneredSpacing {
    static let extraSmall: CGFloat = 4
    static let small: CGFloat = 8
    static let medium: CGFloat = 12
    static let large: CGFloat = 16
    static let extraLarge: CGFloat = 24
    static let huge: CGFloat = 32
    static let massive: CGFloat = 48
}

// TYPOGRAPHY - Sincronizado con Kotlin
struct VeneredTypography {
    static let displayLargeSize: CGFloat = 48
    static let displayMediumSize: CGFloat = 40
    static let displaySmallSize: CGFloat = 32
    
    static let headlineLargeSize: CGFloat = 28
    static let headlineMediumSize: CGFloat = 24
    static let headlineSmallSize: CGFloat = 20
    
    static let titleLargeSize: CGFloat = 20
    static let titleMediumSize: CGFloat = 16
    static let titleSmallSize: CGFloat = 14
    
    static let bodyLargeSize: CGFloat = 16
    static let bodyMediumSize: CGFloat = 14
    static let bodySmallSize: CGFloat = 12
    
    static let labelLargeSize: CGFloat = 14
    static let labelMediumSize: CGFloat = 12
    static let labelSmallSize: CGFloat = 11
}

// CORNER RADIUS - Sincronizado con Kotlin
struct VeneredCornerRadius {
    static let extraSmall: CGFloat = 4
    static let small: CGFloat = 8
    static let medium: CGFloat = 12
    static let large: CGFloat = 16
    static let extraLarge: CGFloat = 20
    static let circle: CGFloat = 50
}

// THEME MANAGER
struct VeneredTheme {
    @Environment(\.colorScheme) var colorScheme
    
    func getPrimaryColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkPrimary : VeneredColors.lightPrimary
    }
    
    func getSecondaryColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkSecondary : VeneredColors.lightSecondary
    }
    
    func getBackgroundColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkBackground : VeneredColors.lightBackground
    }
    
    func getSurfaceColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkSurface : VeneredColors.lightSurface
    }
    
    func getTextPrimaryColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkTextPrimary : VeneredColors.lightTextPrimary
    }
    
    func getTextSecondaryColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkTextSecondary : VeneredColors.lightTextSecondary
    }
    
    func getErrorColor() -> Color {
        colorScheme == .dark ? VeneredColors.darkError : VeneredColors.lightError
    }
}
